pwiz_aux/msrc/utility/vendor_api/thermo/RawFile.cpp

```diff
@@ -1014,6 +1015,54 @@ MassListPtr RawFileImpl::getMassListFromLabelData(long scanNumber)
 }*/
 #endif
 
+NoiseListPtr RawFileImpl::getNoiseList(long scanNumber) const
+{
+    try
+    {
+        auto result = boost::make_shared<NoiseList>();
+
+#ifndef _WIN64
+        if (rawInterfaceVersion_ < 2)
+            throw RawEgg("[RawFileImpl::getNoiseData()] GetNoiseData requires the IXRawfile2 interface.");
+
+        IXRawfile2Ptr raw2 = (IXRawfile2Ptr)raw_;
+        _variant_t varNoisePacket;
+        raw2->GetNoiseData(&varNoisePacket, &scanNumber);
+
+        if (varNoisePacket.vt != (VT_ARRAY | VT_R8))
+            throw RawEgg("getNoiseList(): VARIANT error.");
+
+        _variant_t noisePacket2(varNoisePacket, false);
+        long size = (long)noisePacket2.parray->rgsabound[0].cElements;
+        result->mzArray.resize(size);
+        result->baselineArray.resize(size);
+        result->intensityArray.resize(size);
+        double* pdval = (double*)noisePacket2.parray->pvData;
+        for (long i = 0; i < size; ++i)
+        {
+            result->mzArray[i] = (double)pdval[(i * 3) + 0];
+            result->intensityArray[i] = (float)pdval[(i * 3) + 1];

```

Comment:
Since you're casting here you might as well cast to `double` (and change NoiseData accordingly) because that's what the BinaryDataArrays are so it'll make swapping them with `swapBinaryDataArray()` possible.